custom_components/tahoma/tahoma_device.py

```diff
@@ -27,43 +29,45 @@
 class TahomaDevice(Entity):
     """Representation of a TaHoma device entity."""
 
-    def __init__(self, tahoma_device, controller):
+    def __init__(self, device: Device, client: TahomaClient):
         """Initialize the device."""
-        self.tahoma_device = tahoma_device
-        self.controller = controller
+        self.device = device
+        self.client = client
         self._exec_queue = []
 
-    def update(self):
+    async def async_update(self):
         """Update method."""
-        if self.should_wait():
-            self.hass.add_job(self.update)
+        if await self.should_wait():
+            self.hass.async_add_job(self.async_update)
             return
-        self.controller.get_states([self.tahoma_device])
-        self.schedule_update_ha_state()
+
+        self.device.states = await self.client.get_state(self.device.deviceurl)
 
     async def async_added_to_hass(self):
         """Entity created."""
         await super().async_added_to_hass()
-        self.schedule_update_ha_state(True)
+        self.async_schedule_update_ha_state(True)
 
     @property
     def name(self):
         """Return the name of the device."""
-        return self.tahoma_device.label
+        return self.device.label
 
     @property
     def available(self) -> bool:
         """Return True if entity is available."""
-        states = self.tahoma_device.active_states
+        # TODO Figure out how this works for RTS devices
+        if self.device.available is not None:
```

Comment:
We didn't spot yet that the API also sends back `enabled` and `available` in the device information. We possibly can simplify this module, by removing the 3 if statements.

However I don't know if this `available` property covers all cases below and what it does for RTS devices.

Reply:
@vlebourl could you give this a try? Or pull the new API manually to understand the status of `available` for your RTS devices.